### PR TITLE
[BTS-266] Turning `--cluster.force-one-shard` on later should not always enable Analyzer optimizations

### PR DESCRIPTION
**Pull request by cpjulia**
### Scope & Purpose

When starting up a cluster without `--cluster.force-one-shard`, creating a database that might not be single sharded and then restarting the cluster with the startup option `--cluster.force-one-shard` set to true, when the formerly created database has more than one shard, but the flag is set to true, this could lead to arangosearch's analyzers to use optimizations that should not be used if not in a single shard mode. For the optimizations not to happen for the old database created without the `--cluster.force-one-shard` set to true, the verification of the parameter being true as a condition to run optimizations was removed.

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [x] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [x] **integration tests**
  - [ ] **resilience tests**
- [x] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports 
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

- [ ] Docs PR: 
- [x] Enterprise PR: https://github.com/arangodb/enterprise/pull/1194
- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/BTS-266
- [ ] Design document:

### CODE REVIEW

File: arangod/IResearch/IResearchAnalyzerFeature.cpp [hunk 1]
```diff
@@ -563,6 +562,15 @@ Result visitAnalyzers(TRI_vocbase_t& vocbase,
...
     }
+    /*
+    else if
+    (vocbase.server().getFeature<arangodb::ClusterFeature>().forceOneShard()){
+      TRI_IF_FAILURE("CheckNotForceShardOnDBCreation") {
+        return {TRI_ERROR_INTERNAL, "Optimization will not be made for
+    previously created DB without forceOneShard set"}
+      }
+    }
+     */
```
Comment (MBkkt) [inline arangod/IResearch/IResearchAnalyzerFeature.cpp:573]:
```suggestion
```

File: arangod/IResearch/IResearchAnalyzerFeature.cpp [hunk 2]
```diff
@@ -505,8 +505,7 @@ Result visitAnalyzers(TRI_vocbase_t& vocbase,
     if (!coords.empty() &&
         !vocbase.isSystem() &&  // System database could be on other server so
                                 // OneShard optimization will not work
-        (vocbase.server().getFeature<ClusterFeature>().forceOneShard() ||
-         vocbase.isOneShard())) {
+        (vocbase.isOneShard())) {
```
Comment (MBkkt) [inline arangod/IResearch/IResearchAnalyzerFeature.cpp:508]:
```suggestion
        vocbase.isOneShard()) {
```

**Review by MBkkt — CHANGES_REQUESTED** [review]
Could you add test please?

File: arangod/IResearch/IResearchAnalyzerFeature.cpp (hunk 1 above)
Comment (cpjulia) [reply arangod/IResearch/IResearchAnalyzerFeature.cpp]: Made in enterprise

File: arangod/IResearch/IResearchAnalyzerFeature.cpp [hunk 3]
```diff
@@ -563,6 +573,17 @@ Result visitAnalyzers(TRI_vocbase_t& vocbase,
...
       }
     }
+    TRI_IF_FAILURE("CheckDBWhenNotSingleShardAndFlagChange") {
+      if (vocbase.server()
+              .getFeature<arangodb::ClusterFeature>()
+              .forceOneShard()) {
+        return {TRI_ERROR_DEBUG,
+                "force-one-shard true and db is not single shard"};
+      } else {
+        return {TRI_ERROR_DEBUG,
```
Comment (MBkkt) [inline arangod/IResearch/IResearchAnalyzerFeature.cpp:583]: Why is this case the error? Both false and both true sounds good?

I understand it's for test but still looks strange

File: arangod/IResearch/IResearchAnalyzerFeature.cpp [hunk 4]
```diff
@@ -505,8 +505,18 @@ Result visitAnalyzers(TRI_vocbase_t& vocbase,
...
                                 // OneShard optimization will not work
-        (vocbase.server().getFeature<ClusterFeature>().forceOneShard() ||
-         vocbase.isOneShard())) {
+        vocbase.isOneShard()) {
+      TRI_IF_FAILURE("CheckDBWhenSingleShardAndFlagChange") {
+        if (vocbase.server()
+                .getFeature<arangodb::ClusterFeature>()
+                .forceOneShard()) {
+          return {TRI_ERROR_DEBUG,
+                  "force-one-shard true and db is single shard"};
```
Comment (MBkkt) [inline arangod/IResearch/IResearchAnalyzerFeature.cpp:514]: Same question

File: arangod/IResearch/IResearchAnalyzerFeature.cpp (hunk 3 above)
Comment (MBkkt) [reply arangod/IResearch/IResearchAnalyzerFeature.cpp:583]: I see, but my question about why do you need fail point for case when forceOneShard == isOneShard

**Comment by MBkkt** [comment]
Also this PR probably needs backports?

**Review by MBkkt — APPROVED** [review]

**Review by mchacki — APPROVED** [review]
LGTM

**Comment by Dronplane** [comment]
As a note: this will prevent us from using one-shard optimization for analyzers reading in case of server run with force-one-shard set to true. I'd suggest to make "isOneShard" always return correct value no matter how this "single shard" property was achieved. If that is too much for this PR, let's open a BTS ticket for this, to fix it later on.

**Review by jsteemann — APPROVED** [review]
LGTM

File: js/client/modules/@arangodb/testsuites/server_permissions.js [hunk 5]
```diff
@@ -115,7 +115,7 @@ class permissionsRunner extends tu.runLocalInArangoshRunner {
                                                        paramsFirstRun,
                                                        this.friendlyName,
                                                        rootDir);
-          
+	  global.theInstanceManager = this.instanceManager;
```
Comment (jsteemann) [inline js/client/modules/@arangodb/testsuites/server_permissions.js:118]:
```suggestion
	        global.theInstanceManager = this.instanceManager;
```

**Review by jsteemann — APPROVED** [review]
LGTM

**Pull request merged**